File: IsDbBisewFinalProject/Models/Purchase.cs
using Microsoft.EntityFrameworkCore.Metadata.Internal;
using System.ComponentModel.DataAnnotations.Schema;
using System.ComponentModel.DataAnnotations;
using System.Xml.Linq;
using System;
using System.Collections;
using System.Collections.Generic;

namespace IsDbBisewFinalProject.Models
{
    public class Purchase
    {
        public int Id { get; set; }
      
        [Required, Column(TypeName = "date"), DisplayFormat(DataFormatString = "{0:yyyy-MM-dd}", ApplyFormatInEditMode = true)]
        public DateTime PurchaseDate { get; set; }

        [ForeignKey("Supplier"),Display(Name ="Supplier")]
        public int SupplierId { get; set; }

        [Required,Display(Name ="Total Amount")]
        [Column(TypeName = "money")]
        public decimal TotalAmount { get; set; }

        //nev 
        public virtual Supplier Supplier { get; set; }
        public virtual ICollection<PurchaseDetail> PurchaseDetails { get; set; }
    }
}

File: IsDbBisewFinalProject/ClientApp/src/app/models/purchase/purchase.ts
import { MatTableDataSource } from "@angular/material/table";
import { empty } from "rxjs";
import { PurchaseDetail } from "../purchase-detail/purchase-detail";


export class Purchase {

  constructor(
    // purchase
    public id?: number,
    public purchaseDate?: Date,
    public supplierId?: number,
    public purchaseId?:number,
    public totalAmount?: number,
    public purchaseDetails?: PurchaseDetail[] | MatTableDataSource<PurchaseDetail>
   
  ) { }
}
